plugins/events-node/src/api/EventSubscriber.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Handles received events.
+ * This may include triggering refreshes of catalog entities
+ * or other actions to react on events.
+ *
+ * @public
+ */
+export interface EventSubscriber<EventPayload> {
+  /**
+   * Name of the WebhookEventHandler.
+   * E.g., used at background tasks.
+   */
+  getEventSubscriberName(): string;
+
+  /**
+   * Supported event topics like "github", "bitbucketCloud", etc.
+   */
+  supportsEventTopics(): string[];
+
+  /**
+   * React on a received event.
+   *
+   * @param payload - The event payload.
+   */
+  onEvent(payload: EventPayload): Promise<void>;
```

Comment:
Let us know what you find out. You don't have to change anything right now, just interested in your findings because both approaches probably have merit

Reply:
a quick attempt to replace it using rxjs
https://github.com/Bonial-International-GmbH/backstage/commit/79caff5c358ca06be042d25b5a30c4c46c42d31f

As already discussed: zen-observable is preferred over rxjs in case we move forward with this.

This is only meant as discussion token for now.